extractor/src/main/java/org/schabi/newpipe/extractor/linkhandler/ListLinkHandlerFactory.java

```diff
@@ -1,24 +1,29 @@
 package org.schabi.newpipe.extractor.linkhandler;
 
 import org.schabi.newpipe.extractor.exceptions.ParsingException;
+import org.schabi.newpipe.extractor.search.filter.FilterItem;
 import org.schabi.newpipe.extractor.utils.Utils;
 
-import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
 
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
 public abstract class ListLinkHandlerFactory extends LinkHandlerFactory {
 
     ///////////////////////////////////
     // To Override
     ///////////////////////////////////
 
-    public abstract String getUrl(String id, List<String> contentFilter, String sortFilter)
+    public abstract String getUrl(String id,
+                                  @Nonnull List<FilterItem> contentFilter,
+                                  @Nullable List<FilterItem> sortFilter)
```

Comment:
Shouldn't we just use one array with `FilterItem` and stop making the distinction between content and sort filters? What is the advantage of having two separate arrays?

Reply:
a content filter might have a different set of sort filters.

Reply:
But there can already be content filters from various available filter groups, so why can't sort filters be just another filter group? Maybe I'm missing something

Reply:
I'll have to review my code again. I think I thought about that.